### PR TITLE
hide buttons when showButtons is false && Overlay - Context Menu - Dropdown is not hiding when a click happens in table rows

### LINKED ISSUE #15937: Table: columnFilter
### Describe the bug

when showButtons is false <div class="p-column-filter-buttonbar"> is being rendered

### Environment

17.16.0

### Reproducer

_No response_

### Angular version

17

### PrimeNG version

17.16.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v20.12.1

### Browser(s)

All

### Steps to reproduce the behavior

create custom footer template then there is padding event if showButtons  is false

### Expected behavior

do not render <div class="p-column-filter-buttonbar"> when showButtons is false

### LINKED ISSUE #15939: Overlay - Context Menu - Dropdown is not hiding when a click happens in table row edit mode
### Describe the bug

Overlay - Context Menu - Dropdown is not hiding when a click happens in table row edit mode

### Environment

17.16.0

### Reproducer

_No response_

### Angular version

17

### PrimeNG version

17.16.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v20.12.1

### Browser(s)

All

### Steps to reproduce the behavior

Overlay - Context Menu - Dropdown is not hiding when a click happens in table row edit mode

### Expected behavior

Overlay - Context Menu - Dropdown hide if inside table row edit mode clicked

### PR DESCRIPTION
**Pull request by ymg2006**
fixes #15937
fixes #15939

### CODE REVIEW

**Review by rosenthalj — COMMENTED** [review]
@ymg2006 
@mehmetcetin01140

I quickly reviewed  this PR #15938 because it changes PrimeNG components that are widely used in my project.

Since Issue #15937 and Issue #15939 do not have **reproducers** it makes harder to understand and determine if the changes fixes the issues.

Quickly looking at the changes (and not trying to fully understand the changes),  it appears that the changes address different problems and  should be separated into 2 or 3 separate PRs.  

The change to contextmenu.ts (line 610) modifies the listener from **click** to **mousedown**.   From experience with working on PrimeNG issues,  I know that similar code (to insure similar functionality) is used throughout the PrimeNG code base.   If significant  a change is made to contextmenu then most likely the same change is needed throughout the PrimeNG menu code and possibly other places.  The annotated screenshot listed below shows the results of a search of the PrimeNG code baseline showing the **click** listener is used throughout the PrimeNG and in lots (and probably all) the PrimeNG menu components

Window_and_Code_search_results_and_Code_search_results

File: src/app/components/table/table.ts [hunk 1]
```diff
@@ -5187,7 +5187,7 @@ export class ReorderableRow implements AfterViewInit {
                             <ng-template *ngTemplateOutlet="addRuleIconTemplate"></ng-template>
                         </button>
                     </div>
-                    <div class="p-column-filter-buttonbar">
+                    <div class="p-column-filter-buttonbar" *ngIf="showButtons">
```
Comment (rosenthalj) [inline src/app/components/table/table.ts:5190]: Is this the only change needed to fix issue #15937? I'm assuming this is not needed for issue #15937

Note: i did not try to verify the change

File: src/app/components/contextmenu/contextmenu.ts [hunk 2]
```diff
@@ -607,7 +607,7 @@ export class ContextMenu implements OnInit, AfterContentInit, OnDestroy {
             if (!this.documentClickListener) {
                 const documentTarget: any = this.el ? this.el.nativeElement.ownerDocument : 'document';
 
-                this.documentClickListener = this.renderer.listen(documentTarget, 'click', (event) => {
+                this.documentClickListener = this.renderer.listen(documentTarget, 'mousedown', (event) => {
                     if (this.containerViewChild.nativeElement.offsetParent && this.isOutsideClicked(event) && !event.ctrlKey && event.button !== 2 && this.triggerEvent !== 'click') {
```
Comment (rosenthalj) [inline src/app/components/contextmenu/contextmenu.ts:611]: Line 610 sets listener from **click** to **mousedown**

Line 611 still references **click**.  Is this correct?

**Review by rosenthalj — COMMENTED** [review]
A few code comments

**Comment by ymg2006** [comment]
> @ymg2006 @mehmetcetin01140
> 
> I quickly reviewed this PR #15938 because it changes PrimeNG components that are widely used in my project.
> 
> Since Issue #15937 and Issue #15939 do not have **reproducers** it makes harder to understand and determine if the changes fixe the issues.
> 
> Quickly looking at the changes (and not trying to fully understand the changes), it appears that the changes address different problems and should be separated into 2 or 3 separate PRs.
> 
> The change to contextmenu.ts (line 610) modifies the listener from **click** to **mousedown**. From experience with working on PrimeNG issues, I know that similar code (to insure similar functionality) is used throughout the PrimeNG code base. If s significant change is made to contextmenu.ts then most likely the same change is needed throughout the PrimeNG menu code and possibly other places. The annotated screenshot listed below shows the results of a search of the PrimeNG code baseline showing the **click** listener is used throughout PrimeNG (14 files) and in lots (and probably all) the PrimeNG menu components.
> 
> Window_and_Code_search_results_and_Code_search_results

@rosenthalj put context menu to one p-table and right click on one of them, then click on another row....
it will not close the context menu. exact same issue is reproduced by overlay ...
For the other components, I have not tried to reproduce same issue. I widely used overlay and context menu. drop down is automatically fixed since it uses overlay.

**Comment by ymg2006** [comment]
I will make them separate issues with reproducing sample and create separate pull request for each

**Pull request closed without merging**